### PR TITLE
GetAllRuleCategories API

### LINKED ISSUE #328: [FEATURE] GetAllRuleCategories API
**Is your feature request related to a problem?**
Currently FE has all rule categories hardcoded.

**What solution would you like?**
We need capability for FE to fetch current rule categories from backend. We can implement new API GetAllRuleCategories which will return all categories enabled in backend:

```
GET /_plugins/_security_analytics/rules/categories
```

RESPONSE:
```
{
  "rule_categories": [
    {
      "key": "ad_ldap",
      "display_name": "AD/LDAP"
    },
    {
      "key": "dns",
      "display_name": "DNS logs"
    },
    {
      "key": "network",
      "display_name": "Network"
    },
    {
      "key": "apache_access",
      "display_name": "Apache access logs"
    },
    {
      "key": "cloudtrail",
      "display_name": "Cloud Trail logs"
    },
    {
      "key": "s3",
      "display_name": "S3 access logs"
    },
    {
      "key": "windows",
      "display_name": "Windows logs"
    },
    {
      "key": "linux",
      "display_name": "System logs"
    }
  ]
}
```

Response fields:
| field | description |
| --- | --- |
| **_key_** | This value should be used as detectorType/ruleCategory param when calling APIs.(CreateDetector, SearchRules,...)  |
| **_display_name_**  | Used on UI as label text |

### PR DESCRIPTION
**Pull request by petardz**
### Description
API to fetch all available log types/rule categories
 
### Issues Resolved
 #328

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/model/RuleCategory.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    static {
+        List<RuleCategory> ruleCategories = new ArrayList<>();
+        ruleCategories.add(new RuleCategory(Detector.DetectorType.AD_LDAP.getDetectorType(), "AD/LDAP", true));
+        ruleCategories.add(new RuleCategory(Detector.DetectorType.DNS.getDetectorType(), "DNS logs", true));
+        ruleCategories.add(new RuleCategory(Detector.DetectorType.NETWORK.getDetectorType(), "Netflow", true));
+        ruleCategories.add(new RuleCategory(Detector.DetectorType.APACHE_ACCESS.getDetectorType(), "Apache access logs", true));
+        ruleCategories.add(new RuleCategory(Detector.DetectorType.CLOUDTRAIL.getDetectorType(), "Cloud Trail logs", true));
+        ruleCategories.add(new RuleCategory(Detector.DetectorType.S3.getDetectorType(), "S3 access logs", true));
+        ruleCategories.add(new RuleCategory(Detector.DetectorType.WINDOWS.getDetectorType(), "Windows logs", true));
+        ruleCategories.add(new RuleCategory(Detector.DetectorType.LINUX.getDetectorType(), "System logs", true));
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/securityanalytics/model/RuleCategory.java:65]: Can we define these in form of a resource (config) file externally?

Comment (petardz) [reply src/main/java/org/opensearch/securityanalytics/model/RuleCategory.java]: done

**Review by sbcd90 — APPROVED** [review]

File: src/main/resources/rules/rule_categories.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "rule_categories": [
```
Comment (getsaurabh02) [inline src/main/resources/rules/rule_categories.json:2]: Have we started using this in the frontend yet? If not lets create an issue on frontend to adopt this. 
cc : @amsiglan

**Review by getsaurabh02 — APPROVED** [review]
LGTM

**Pull request merged**